app/server/utils/user.py

```diff
@@ -592,34 +601,24 @@ def send_phone_verification_message(to_phone, one_time_code):
         send_message(to_phone, reciever_message)
 
 
-def change_pin(user: User, new_pin):
-    user.pin = new_pin
-    db.session.commit()
-    send_pin_change_messages(user, new_pin)
+def send_sms(user, message_key):
+    message = i18n_for(user, "ussd.kenya.{}".format(message_key))
+    send_message(user.phone, message)
 
 
-def change_initial_pin(user: User, new_pin):
-    user.pin = new_pin
-    user.is_activated = True
-    # TODO [Philip]: Find out whether reset token should be deleted.
-    db.session.commit()
-    send_pin_change_messages(user, new_pin)
-
+def change_pin(user, new_pin):
+    try:
+        user.pin = new_pin
+        db.session.commit()
+        send_sms(user, 'successful_pin_change_sms')
+    except InvalidRequestError:
```

Comment:
this is the error that happens from a failed db.session.commit?

Reply:
Yes that's the error it throws.